src/main/java/org/myrobotlab/service/MqttBroker.java

```diff
@@ -0,0 +1,628 @@
+ * /mrl/clock01/setInterval with payload ['2000']
+ * 
+ * this will invoke clock01.setInterval(Integer ms) Make note: the parameter
+ * list values are encoded, that is why its '2000' vs 2000, because first the
+ * whole array is decoded - then each parameter is, similar to http form values
+ *
+ */
+public class MqttBroker extends Service implements InterceptHandler, Gateway {
+
+  public final static Logger log = LoggerFactory.getLogger(MqttBroker.class);
+
+  private static final long serialVersionUID = 1L;
+
+  private static String bytesToHex(byte[] hash) {
+    StringBuilder hexString = new StringBuilder(2 * hash.length);
+    for (int i = 0; i < hash.length; i++) {
+      String hex = Integer.toHexString(0xff & hash[i]);
+      if (hex.length() == 1) {
+        hexString.append('0');
+      }
+      hexString.append(hex);
+    }
+    return hexString.toString();
+  }
+
+  private static final MqttMsg toMqttMsg(InterceptMessage msg) {
+    MqttMsg retMsg = new MqttMsg();
+    if (msg instanceof InterceptPublishMessage) {
+      InterceptPublishMessage m = (InterceptPublishMessage) msg;
+      retMsg.setClientID(m.getClientID());
+      retMsg.setUsername(m.getUsername());
+      retMsg.setTopicName(m.getTopicName());
+      ByteBuf b = m.getPayload();
+      b.hasArray();
+      String payload = b.hasArray() ? new String(m.getPayload().array(), UTF_8) : b.toString(UTF_8);
+      retMsg.setPayload(payload);
+
+    } else if (msg instanceof InterceptAcknowledgedMessage) {
+      InterceptAcknowledgedMessage m = (InterceptAcknowledgedMessage) msg;
+      retMsg.setUsername(m.getUsername());
+
+    } else if (msg instanceof InterceptConnectionLostMessage) {
+      InterceptConnectionLostMessage m = (InterceptConnectionLostMessage) msg;
+      retMsg.setClientID(m.getClientID());
+      retMsg.setUsername(m.getUsername());
+
+    } else if (msg instanceof InterceptConnectMessage) {
+      InterceptConnectMessage m = (InterceptConnectMessage) msg;
+      retMsg.setClientID(m.getClientID());
+      retMsg.setUsername(m.getUsername());
+
+    } else if (msg instanceof InterceptDisconnectMessage) {
+      InterceptDisconnectMessage m = (InterceptDisconnectMessage) msg;
+      retMsg.setClientID(m.getClientID());
+      retMsg.setUsername(m.getUsername());
+
+    } else if (msg instanceof InterceptSubscribeMessage) {
+      InterceptSubscribeMessage m = (InterceptSubscribeMessage) msg;
+      retMsg.setClientID(m.getClientID());
+      retMsg.setUsername(m.getUsername());
+      retMsg.setTopicName(m.getTopicFilter());
+
+    } else if (msg instanceof InterceptUnsubscribeMessage) {
+      InterceptUnsubscribeMessage m = (InterceptUnsubscribeMessage) msg;
+      retMsg.setClientID(m.getClientID());
+      retMsg.setUsername(m.getUsername());
+      retMsg.setTopicName(m.getTopicFilter());
+    }
+    return retMsg;
+  }
+
+  protected String address = "0.0.0.0";
+
+  protected Boolean allow_anonymous = false;

```

Comment:
why use objects when primitive would be better and less ambigious.  boolean is true or false.. Boolean is true,false or null.  Seems like opting for primitives here will avoid a bunch of null pointer exeptions

Reply:
moquette embedded mqtt server modeled its goofy config form mosquittos goofy config which had this boolean vs having username/pwd non empty or multiple credentials - which should have been the source of the true/false ..

not used - removing ....